jfrog-oauth/main.test.ts

```diff
@@ -1,19 +1,115 @@
   testRequiredVariables,
 } from "../test";
 
 describe("jfrog-oauth", async () => {
   await runTerraformInit(import.meta.dir);
 
-  testRequiredVariables(import.meta.dir, {
-    agent_id: "some-agent-id",
-    jfrog_url: "http://localhost:8081",
-    package_managers: "{}",
+  const fakeFrogHostAndPort = "localhost:8081";
+  const fakeFrogUrl = `http://${fakeFrogHostAndPort}`;
+
+  it("can run apply with required variables", async () => {
+    testRequiredVariables(import.meta.dir, {
+      agent_id: "some-agent-id",
+      jfrog_url: fakeFrogUrl,
+      package_managers: "{}",
+    });
   });
-});
 
-//TODO add more tests
+  it("generates an npmrc with scoped repos", async () => {
+    const state = await runTerraformApply(import.meta.dir, {
+      agent_id: "some-agent-id",
+      jfrog_url: fakeFrogUrl,
+      package_managers: JSON.stringify({
+        npm: ["global", "@foo:foo", "@bar:bar"],
+      }),
+    });
+    const coderScript = findResourceInstance(state, "coder_script");
+    const npmrcStanza = `cat << EOF > ~/.npmrc
+email=default@example.com
+registry=${fakeFrogUrl}/artifactory/api/npm/global
+//${fakeFrogHostAndPort}/artifactory/api/npm/global/:_authToken=
+@foo:registry=${fakeFrogUrl}/artifactory/api/npm/foo
+//${fakeFrogHostAndPort}/artifactory/api/npm/foo/:_authToken=
+@bar:registry=${fakeFrogUrl}/artifactory/api/npm/bar
+//${fakeFrogHostAndPort}/artifactory/api/npm/bar/:_authToken=
+
+EOF`;
+    expect(coderScript.script).toContain(npmrcStanza);
+    expect(coderScript.script).toContain(
+      'jf npmc --global --repo-resolve "global"',
+    );
+    expect(coderScript.script).toContain(
+      'if [ -z "YES" ]; then\n  not_configured npm',
+    );
+  });
+
+  it("generates a pip config with extra-indexes", async () => {
+    const state = await runTerraformApply(import.meta.dir, {
+      agent_id: "some-agent-id",
+      jfrog_url: fakeFrogUrl,
+      package_managers: JSON.stringify({
+        pypi: ["global", "foo", "bar"],
+      }),
+    });
+    const coderScript = findResourceInstance(state, "coder_script");
+    const pipStanza = `cat << EOF > ~/.pip/pip.conf
+[global]
+index-url = https://default:@${fakeFrogHostAndPort}/artifactory/api/pypi/global/simple
+extra-index-url =
+    https://default:@${fakeFrogHostAndPort}/artifactory/api/pypi/foo/simple
+    https://default:@${fakeFrogHostAndPort}/artifactory/api/pypi/bar/simple
+
+EOF`;
+    expect(coderScript.script).toContain(pipStanza);
+    expect(coderScript.script).toContain(
+      'jf pipc --global --repo-resolve "global"',
+    );
+    expect(coderScript.script).toContain(
+      'if [ -z "YES" ]; then\n  not_configured pypi',
+    );
+  });
+
+  it("registers multiple docker repos", async () => {
+    const state = await runTerraformApply(import.meta.dir, {
+      agent_id: "some-agent-id",
+      jfrog_url: fakeFrogUrl,
+      package_managers: JSON.stringify({
+        docker: ["foo.jfrog.io", "bar.jfrog.io", "baz.jfrog.io"],
+      }),
```

Comment:
This definitely isn't your fault, but I'm making a note so I don't forget: we should update our helper functions so that they support Terraform objects directly